ml-completion/src/main/kotlin/RustElementFeatureProvider.kt

```diff
@@ -0,0 +1,197 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ml
+
+import com.intellij.codeInsight.completion.CompletionLocation
+import com.intellij.codeInsight.completion.ml.ContextFeatures
+import com.intellij.codeInsight.completion.ml.ElementFeatureProvider
+import com.intellij.codeInsight.completion.ml.MLFeatureValue
+import com.intellij.codeInsight.lookup.LookupElement
+import com.intellij.psi.PsiElement
+import org.rust.ide.utils.import.isStd
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.RsElement
+import org.rust.lang.core.psi.ext.containingCrate
+
+@Suppress("UnstableApiUsage")
+class RustElementFeatureProvider : ElementFeatureProvider {
+    override fun getName(): String = "rust"
+
+    override fun calculateFeatures(
+        element: LookupElement,
+        location: CompletionLocation,
+        contextFeatures: ContextFeatures
+    ): Map<String, MLFeatureValue> {
+        val result = hashMapOf<String, MLFeatureValue>()
+        val psiElement = element.psiElement
+        val lookupString = element.lookupString
+
+        /** If [element] is a keyword, store keyword kind as "kind" feature and finish */
+        val keywordKind = RsKeywordMLKind.from(lookupString)
+        if (keywordKind != null) {
+            result["kind"] = MLFeatureValue.categorical(keywordKind)
+            return result
+        }
+
+        /**
+         * Otherwise, if [element] is [RsElement],
+         * store PSI kind as "kind" feature
+         * and store if its origin is stdlib as "is_from_stdlib" feature
+         */
+        val rsElement = psiElement as? RsElement ?: return result
+        val rsPsiElementKind = RsPsiElementMLKind.from(rsElement)
+        if (rsPsiElementKind != null) {
+            result["kind"] = MLFeatureValue.categorical(rsPsiElementKind)
+        }
+        val containingCrate = rsElement.containingMod.containingCrate
+        if (containingCrate != null) {
+            result["is_from_stdlib"] = MLFeatureValue.binary(containingCrate.isStd)
+        }
+
+        return result
+    }
+}
+
+/** See [org.rust.lang.core.psi.RsTokenType] */
+internal enum class RsKeywordMLKind {
```

Comment:
Let's somehow check that this list contains exactly the same elements as `org.rust.lang.core.psi.RsTokenTypeKt#RS_KEYWORDS` not to forgot to add the corresponding element to the second list when you add a new keyword element

Reply:
Added `test all keywords are covered` test to `RsElementFeatureProviderTest`